vowpalwabbit/core/tests/io_alignment_test.cc

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+#include "vw/core/io_buf.h"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+using namespace testing;
+
+using align_t = VW::desired_align::align_t;
+using flatbuffer_t = VW::desired_align::flatbuffer_t;
+using VW::desired_align;
+
+struct positioned_ptr
+{
+  size_t allocation;
+  void* allocation_unit;
+  int8_t* p;
+  static_assert(sizeof(int8_t) == 1, "int8_t is not 1 byte");
```

Comment:
crazy that this is needed! :)

Reply:
Not strictly necessary. Only around if someone is playing fast and loose with typedefs/usings.